x/farming/keeper/plan.go

```diff
@@ -213,24 +250,66 @@ func (k Keeper) CreateRatioPlan(ctx sdk.Context, msg *types.MsgCreateRatioPlan,
 	return ratioPlan, nil
 }
 
-// TerminatePlan sends all remaining coins in the plan's farming pool to
-// the termination address and mark the plan as terminated.
+// TerminatePlan marks the plan as terminated.
+// It moves the plan under different store key, which is for terminated plans.
 func (k Keeper) TerminatePlan(ctx sdk.Context, plan types.PlanI) error {
+	_ = plan.SetTerminated(true)
+	k.SetPlan(ctx, plan)
+	if plan.GetType() == types.PlanTypePrivate {
+		k.DecrementNumPrivatePlans(ctx, 1)
+	}
```

Comment:
We don't need to remove the public plans when terminate?

Reply:
Modified the code to delete public plans directly in `TerminatePlan`.